### PR TITLE
[risk=low][RW-6159] Reload Recent Resources on home page after deleting a workspace

### PR DESCRIPTION
**Pull request by jmthibault79**
If you delete a Workspace on the homepage, using the Workspace Card context menu, the Recent Resources list doesn't update, so it's possible to have resource entries that refer to a deleted workspace or resource.  Attempting to access these leads to bad UX: the app just appears to hang.

This PR causes the Recent Resources list to refresh after deleting a workspace, ensuring that these links do not appear.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [x] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/resource-list.tsx [hunk 1]
```diff
@@ -108,10 +108,6 @@ interface Props {
 export const ResourceList = fp.flow(withCdrVersions())((props: Props) => {
   const [tableData, setTableData] = useState<TableData[]>();
 
-  const reloadResources = async () => {
```
Comment (jmthibault79) [inline ui/src/app/components/resource-list.tsx:111]: unrelated, just removing an indirection that I didn't find to be helpful

File: ui/src/app/pages/homepage/homepage.tsx [hunk 2]
```diff
@@ -306,7 +306,7 @@ export const Homepage = fp.flow(
             <FadeBox style={styles.fadeBox}>
               {/* The elements inside this fadeBox will be changed as part of ongoing homepage redesign work */}
               <FlexColumn style={{ justifyContent: 'flex-start' }}>
-                <Workspaces />
+                <Workspaces onChange={() => this.loadWorkspaces()} />
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/homepage.tsx:309]: How this works:
`loadWorkspaces()` updates the state of `userWorkspacesResponse` which is what `RecentResources` uses to populate its table

File: ui/src/app/pages/homepage/homepage.tsx [hunk 3]
```diff
@@ -285,15 +285,15 @@ export const Homepage = fp.flow(
...
+    async loadWorkspaces() {
       return fetchWithSystemErrorHandler(() =>
         workspacesApi().getWorkspaces()
       ).then((response) => this.setState({ userWorkspacesResponse: response }));
     }
 
     userHasWorkspaces(): boolean {
       const { userWorkspacesResponse } = this.state;
-      return userWorkspacesResponse && userWorkspacesResponse.items.length > 0;
+      return userWorkspacesResponse?.items?.length > 0;
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/homepage.tsx:296]: misc cleanup, not strictly related

File: ui/src/app/pages/homepage/homepage.tsx [hunk 4]
```diff
@@ -285,15 +285,15 @@ export const Homepage = fp.flow(
       }
     }
 
-    async checkWorkspaces() {
+    async loadWorkspaces() {
```
Comment (tarekmamdouh) [inline ui/src/app/pages/homepage/homepage.tsx:288]: just a suggestion, rename this to fetchWorkspaces, to differentiate the behavior from the loadWorkspaces function in recent-workspaces

**Review by tarekmamdouh — APPROVED** [review]
LGTM

**Pull request merged**